h2o-bindings/bin/custom/python/gen_gam.py

```diff
@@ -15,6 +15,12 @@ def Lambda(self):
     def Lambda(self, value):
         self._parms["lambda"] = value
 
+    def _additional_used_columns(self, parms):
+        """
+        :return: Gam columns if specified.
+        """
+        return parms["gam_columns"]
+
 extensions = dict(
```

Comment:
I don't understand the use of addtional_used_columns.  What is the term used for?

Reply:
if not used the client (python) will automatically add any cols not in X,Y,weight etc. to ignored_columns

Reply:
It is fine for the gam_columns to be included in the ignored_columns if they are not specified as a predictor.  The reason is because in this case, the gamified columns derived from the gam_column s are used as predictors.

Reply:
I added this more for consistency sake, the client should not mark gam_columns as ingored

Reply:
I actually have to manually add gam_columns that are not used as predictors into ignored_columns in Java.  See testModelScoring in GamPipingTest.

Reply:
that is probably unnecessary and could be removed